tutorials/3-intro-to-brian-simulations.ipynb

```diff
@@ -0,0 +1,616 @@
+    "dn/dt = 0.032*(mV**-1)*(15.*mV-v+VT)/\n",
+    "    (exp((15.*mV-v+VT)/(5.*mV))-1.)/ms*(1.-n)-.5*exp((10.*mV-v+VT)/(40.*mV))/ms*n : 1\n",
+    "dh/dt = 0.128*exp((17.*mV-v+VT)/(18.*mV))/ms*(1.-h)-4./(1+exp((40.*mV-v+VT)/(5.*mV)))/ms*h : 1\n",
+    "I : amp (shared) # everything is the same except we've added this shared\n",
+    "C : farad\n",
+    "'''\n",
+    "group = NeuronGroup(N, eqs_HH_3,\n",
+    "                    threshold='v > -40*mV',\n",
+    "                    refractory='v > -40*mV',\n",
+    "                    method='exponential_euler')\n",
+    "group.v = El\n",
+    "group.C = array([0.8, 1, 1.2])*ufarad*cm**-2*area\n",
+    "statemon = StateMonitor(group, 'v', record=True)\n",
+    "group.run_regularly('I = rand()*50*nA', dt=10*ms)\n",
+    "run(50*ms)\n",
+    "figure(figsize=(9, 4))\n",
+    "for i in range(5):\n",
+    "    axvline(i*10, ls='--', c='k')\n",
+    "axhline(El/mV, ls='-', c='lightgray', lw=3)\n",
+    "plot(statemon.t/ms, statemon.v.T/mV, '-')\n",
+    "xlabel('Time (ms)')\n",
+    "ylabel('v (mV)');"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Ahh, that's more like it!\n",
+    "\n",
+    "## Adding input\n",
+    "\n",
+    "Now let's think about a neuron being driven by a sinusoidal input. Let's go back to a leaky integrate-and-fire to simplify the equations a bit."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "start_scope()\n",
+    "A = 2.5\n",
+    "f = 10*Hz\n",
+    "tau = 5*ms\n",
+    "eqs = '''\n",
+    "dv/dt = (I-v)/tau : 1\n",
+    "I = A*sin(2*pi*f*t) : 1\n",
+    "'''\n",
+    "G = NeuronGroup(1, eqs, threshold='v>1', reset='v=0', method='euler')\n",
+    "M = StateMonitor(G, variables=True, record=True)\n",
+    "run(200*ms)\n",
+    "plot(M.t/ms, M.v[0], label='v')\n",
+    "plot(M.t/ms, M.I[0], label='I')\n",
+    "xlabel('Time (ms)')\n",
+    "ylabel('v')\n",
+    "legend(loc='best');"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "So far, so good and the sort of thing we saw in the first tutorial. Now, what if that input current were something we had recorded and saved in a file? In that case, we can use `TimedArray`. Let's start by reproducing the picture above but using `TimedArray`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "start_scope()\n",
+    "A = 2.5\n",
+    "f = 10*Hz\n",
+    "tau = 5*ms\n",
+    "# Create a TimedArray and set the equations to use it\n",
+    "t_recorded = arange(int(200*ms/defaultclock.dt))*defaultclock.dt\n",
+    "I_recorded = TimedArray(A*sin(2*pi*f*t_recorded), dt=defaultclock.dt)\n",
+    "eqs = '''\n",
+    "dv/dt = (I-v)/tau : 1\n",
+    "I = I_recorded(t) : 1\n",
+    "'''\n",
+    "G = NeuronGroup(1, eqs, threshold='v>1', reset='v=0', method='euler')\n",
```

Comment:
Given that the `TimedArray` is constant over the time step, we could actually use `method='linear'` here (for the next example as well) -- on the other hand, changing it from the previous example (with `sin`) might add some noise.